### ISSUE TITLE
FamixTWithInheritance >> #subclassHierarchy not working for ParameterizableClass

### ISSUE THREAD

**Issue by NicolasAnquetil** [moosetechnology/Famix#571]
#subclassHierarchy on a ParameterizableClass should return also the subclasses of its ParameterizedTypes and this is not the case

**Comment by NicolasAnquetil** [comment]
Interestingly, #superclassHierarchy is working from ParameterizedClass (ie. goes to the ParameterizableClass)

This as to do with the way #allSubclassesDo: and #allSuperclassesDo: are implemented

**Comment by Gabriel-Darbord** [comment]
> Interestingly, #superclassHierarchy is working from ParameterizedClass (ie. goes to the ParameterizableClass)
> 
> This as to do with the way #allSubclassesDo: and #allSuperclassesDo: are implemented

That doesn't seem to be the case...
Should this be a new feature?
```st
class := FamixJavaParameterizableClass new name: 'class'.
type := FamixJavaParameterizedType new parameterizableClass: class; name: 'type'.
FamixJavaInheritance new subclass: type; superclass: (FamixJavaClass new name: 'super').
FamixJavaInheritance new subclass: (FamixJavaClass new name: 'sub'); superclass: type.

class subclassHierarchy. "an OrderedCollection()"
class superclassHierarchy "an OrderedCollection()"
```

**Comment by Gabriel-Darbord** [comment]
Due to my bad understanding of ParameterizableClass, this was wrongfully closed with a "fix".
This should now have been fixed for real by #589: ParameterizableClass>>#subclassHierarchy returns also the subclasses of its ParameterizedTypes.
